### PR TITLE
E2E Tests: Update CI build matrix

### LINKED ISSUE #5338: E2E Tests: Update CI build matrix
## Task Description

It would make sense to run the tests against both the latest version of WordPress (5.5 / 5.6 at the time of writing, as well as the oldest supported version (5.3 at the time of writing).

Percy snapshots would probably only need to be taken for the latest one.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Update the e2e tests to run in latest and 5.4.

## Relevant Technical Choices

After much testing, updating the support matrix to support 5.3 didn't work. There are a number of issues with 5.3 that need to be fixed and conditional logic added. 

## Testing Instructions

---

Fixes #5338

### CODE REVIEW

**Comment by spacedmonkey** [comment]
There appears to be an php notice with 5.3. I will try and fix it.

File: .github/workflows/continuous-integration-e2e.yml [hunk 1]
```diff
@@ -9,12 +9,13 @@ on:
...
 jobs:
   e2e:
-    name: ${{ matrix.browser }}
+    name: '${{ matrix.browser }} - WP ${{ matrix.wp }}'
     runs-on: ubuntu-latest
     strategy:
       matrix:
         # TODO: add back Firefox once support is more mature.
         browser: ['chrome']
+        wp: ['latest', '5.4']
```
Comment (spacedmonkey) [inline .github/workflows/continuous-integration-e2e.yml]:
```suggestion
        wp: ['latest', '5.3.4']
```

File: .github/workflows/continuous-integration-e2e.yml [hunk 2]
```diff
@@ -81,12 +82,18 @@ jobs:
...
+      - name: Run E2E tests with percy
         run: npm run test:e2e:percy
         env:
           PERCY_TOKEN: ${{ secrets.PERCY_TOKEN_E2E }}
           PUPPETEER_PRODUCT: ${{ matrix.browser }}
+        if: ${{ matrix.wp == 'latest' }}
+
+      - name: Run E2E tests
+        run: npm run test:e2e
+        if: ${{ matrix.wp != 'latest' }}
```
Comment (swissspidy) [inline .github/workflows/continuous-integration-e2e.yml]: Nit: in `continuous-integration-unit-php.yml` we use a separate `coverage` flag for a similar purpose:

https://github.com/google/web-stories-wp/blob/afb40799a5b2db28a55dda90dd7596706aac0268/.github/workflows/continuous-integration-unit-php.yml#L29-L33

https://github.com/google/web-stories-wp/blob/afb40799a5b2db28a55dda90dd7596706aac0268/.github/workflows/continuous-integration-unit-php.yml#L110-L120

Perhaps we could do the same here too with a `snapshot` flag for readability?

File: .github/workflows/continuous-integration-e2e.yml [hunk 3]
```diff
@@ -9,12 +9,13 @@ on:
 
 jobs:
   e2e:
-    name: ${{ matrix.browser }}
+    name: '${{ matrix.browser }} - WP ${{ matrix.wp }}'
```
Comment (swissspidy) [inline .github/workflows/continuous-integration-e2e.yml:12]: Note to myself to update GitHub status check settings once this merged.

File: .github/workflows/continuous-integration-e2e.yml (hunk 2 above)
Comment (swissspidy) [reply .github/workflows/continuous-integration-e2e.yml]: I see you named it `coverage` as well now

Comment (spacedmonkey) [reply .github/workflows/continuous-integration-e2e.yml]: Sorry, I forgot that part when I was re-doing this. Should I change it to `snapshot`?

**Review by swissspidy — APPROVED** [review]

**Pull request merged**